ocaml/util/xapi_version.mli

```diff
@@ -23,3 +23,5 @@ val git_id : string
 val xapi_version_major : int
 
 val xapi_version_minor : int
+
+val compare_version : string -> string -> int
```

Comment:
It might be interesting to add a small datastructure for xapi versions and use it here, forcing users to convert the versions first:

```ocaml
type version = {major: int; minor: int; build: string}

val of_string : string -> version

val compare : version -> version -> int
```

Reply:
Hi @psafont , I think adding a type for the "xapi version" is a good idea. But I feel it is not easy to change it clearly as:
1. now the name of the file is `xapi_version.ml`,
2. we already have "version" field in `xapi_version.ml`.
3. besides `version`, `xapi_version_major`, `xapi_version_minor`, `git_id`, there are also other field in the `xapi_version.ml`

To change it clearly, I would like to rename `xapi_version.ml` to `version.ml`, and add a type `xapi_version` as:
type xapi_version = {major: int; minor: int; build: string}
But this would need to change all the xapi codes using `Xapi_version`.

What's your though on this, or if current code is acceptable, can we keep it unchanged? Thank you!

Reply:
Yes, that's too much churn for dubious benefit, thanks for looking into it!